kong/resty/dns/client.lua

```diff
@@ -619,6 +619,13 @@ _M.init = function(options)
   options.search = options.search or resolv.search or { resolv.domain }
   log(DEBUG, PREFIX, "search = ", table_concat(options.search,", "))
 
+  -- check if there is special domain like "."
+  for i = 1, #options.search do
+    if options.search[i] == "." then
+      options.search[i] = ""
```

Comment:
What does this do? Should we instead `table.remove` it?

Reply:
dns.client will init by a options table, which has a `search` field. 
It  comes from `etc/reslov.conf`, someting like:
`search xxx.com xxx.org`
then when we call `toip()` `reslove()`, dns.client will try to concat the qname with `search` table in function `search_iter()`,
to get the proper fqdn, such as `qname.xxx.com` `qname.xxx.org`.

if the `search` field is not a normal domain name but a single `.`,
then the string concat will get the wrong result,
it means `concat(qname, '.', '.')` => `qname..` .

`.` is the root domain name, if we find `.`,
we should only add a `.` suffix to the qname instead of concat,
so the right name will be `qname.`

Reply:
Also, `qname.` seems to have the same effect of `qname`,
maybe just remove `.` is another choice.